### DISCUSSION TITLE
Issue with Vercel Platform after CLI Update

### DISCUSSION THREAD

**Discussion by ebceu4** [vercel/community#1599] [Help]
### Summary

Hello Vercel Community

I'm writing to report an issue with my API after a recent update on the Vercel CLI. After some runtime environment changes on how Node is executing on the platform, my API's started throwing the following error:

"To load an ES module, set 'type': 'module' in the package.json or use the .mjs extension."

I tried changing the type to module, but that escalated to other errors, (as using explicit .js extensions when importing, and many more) and it seems like the whole code is not working anymore. I also tried tweaking the 'engines' in my 'package.json' with no luck, but then I realised that the issue was with the Vercel CLI version itself.

When I downgraded the Vercel CLI from the latest version 28.16.4 to 25.1.0, my code started working again without any changes via the Vercel dev command. **My question now is: _how can I specify the CLI version used during the build and runtime on the platform?_**

My project is basic vercel serveless fuctions written in TS based on plain "api" folder template, that was created before fancy new next.js and stuff.

I would greatly appreciate any guidance on this so I can ensure my API works correctly on the Vercel platform without rewriting the code.

Thank you for your help!

**Comment by amyegan** [comment]
Hi Yuriy! You may need to change your TypeScript config. Support for ESM in TS was recently added, and the addition revealed some config problems that were previously ignored. 

The [recommended solution](https://github.com/orgs/vercel/discussions/1225#discussioncomment-4642381) is to change the [module](https://www.typescriptlang.org/tsconfig#module) value in `tsconfig.json`. There are a couple of options:
1) Set `"module": "esnext"` (or `"module": "nodenext"`) in `tsconfig.json`  AND have `package.json` with `"type": "module"`. (Don't forget to follow the [TS ESM rules](https://www.typescriptlang.org/docs/handbook/esm-node.html) and include the file extension on imports)
1) Use CommonJS instead of ES Modules by setting `"module": "commonjs"` in your `tsconfig.json` file 

If you still have problems after updating the config, please share the contents of your `package.json` and `tsconfig.json` files, the runtime error message (or deployment url so we can look it up), the contents of the file that failed (noted in the error message).

You can find more information in https://github.com/vercel/community/discussions/1225#discussioncomment-4642381
